### ISSUE TITLE
Typing backspace after a "word: true" expansion prevents future "word: true" triggers from matching

### ISSUE THREAD

**Issue by Adam13531** [espanso/espanso#1643]
**Description**

You can't chain multiple `word: true` triggers easily.

**Repro steps**

1. Save this as `espanso/match/repro.yml`:

```yml
matches:
  - trigger: "a"
    replace: "b"
    word: true
```

2. Type `a` → space → backspace → `a` → space

**Expected behavior**

The resulting text is `bb `.

**Actual behavior**

The resulting text is `ba `.

**Screenshots**

https://github.com/espanso/espanso/assets/7192897/802179e8-ff52-4ba6-b1d9-b7ea9a2c7e90

**Logs**
Nothing is emitted to the log.

**Setup information**

- OS: macOS Ventura 13.4.1.
- Version: Espanso 2.1.8

**Additional information**

- My real-world scenario was: I was attempting to type out a hotkey string like `⌥⌘⇧R`, for which I want to type `'alt` → space → backspace → `'cmd` → space → backspace → `'shift` → space → backspace → `R`.
- I define almost all of my matches with `word: true` so that I don't have to worry about triggers being substrings of one another. E.g. with triggers for `'n`, `'no`, and `'now`, I would only be able to activate `'n` without `word: true` (or a delay, presumably).
- As a workaround, I have to press `←` and then `→` to clear the internal buffer after each backspace.
- This is similar to #1496.

**Comment by Vistaus** [comment]
Can't reproduce this on Linux, so this might be a Mac-specific bug.

**Comment by smeech** [comment]
How do you get around the fact that \<backspace> undoes the expansion?
I suppose most of us would deal with this situation with a suffixed trigger like "a:" and not use `word: true` when we know we're unlikely to want a space (or other word separator) left.

**Comment by Adam13531** [comment]
> How do you get around the fact that <backspace> undoes the expansion?

I have `undo_backspace: false` in my config file, that way I _can't_ undo expansions.

**Comment by smeech** [comment]
> > How do you get around the fact that  undoes the expansion?
> 
> I have `undo_backspace: false` in my config file, that way I _can't_ undo expansions.

May I ask why?

**Comment by Adam13531** [comment]
> May I ask why?

Of course! I started using Espanso roughly when I filed this bug, so all I remember is that I was frustrated with `undo_backspace` enabled, but I don't fully recall what was annoying about it. 😢 I know that I want a `" "` to be a completion character, so I think that sometimes I would make a typo, press backspace, and end up being in a state where I'd have to know the typo that I made to know whether I have to undo a completion or not.

I do end up having to work around #1643 several times per week, which means I end up sprinkling in many more arrows keys than I'd like as I type regular sentences.

EDIT: an emoji somehow got moved by an entire sentence. 👀

**Comment by smeech** [comment]
Gotcha. I understand. For a while, I used the misspellings package, which was very similar to an AutoHotKey package colleagues and I used at work.  But I'm a pretty good speller mostly, so I'd often be backspacing to correct what I'd typoed before I had realised that Espanso had corrected it. I've abandoned the package - my typing is much better than it was twenty years ago!
Your use-case is pretty narrow, as most of us would use other ways of dealing with the issue, but I'll leave it up as it's clearly a minor bug. Whether it'll get any attention is another matter.
